tests/baselines_test/cmame_test.py

```diff
@@ -0,0 +1,128 @@
+"""Tests CMA ME implementation"""
+
+from typing import Dict, Tuple, Type
+
+import jax
+import jax.numpy as jnp
+import pytest
+
+from qdax.core.containers.mapelites_repertoire import (
+    MapElitesRepertoire,
+    compute_euclidean_centroids,
+)
+from qdax.core.emitters.cma_emitter import CMAEmitter
+from qdax.core.emitters.cma_opt_emitter import CMAOptimizingEmitter
+from qdax.core.emitters.cma_pool_emitter import CMAPoolEmitter
+from qdax.core.emitters.cma_rnd_emitter import CMARndEmitter
+from qdax.core.map_elites import MAPElites
+from qdax.types import Descriptor, ExtraScores, Fitness, RNGKey
+
+
+@pytest.mark.parametrize(
+    "emitter_type",
+    [CMAOptimizingEmitter, CMARndEmitter, CMAEmitter],
+)
+def test_cma_me(emitter_type: Type[CMAEmitter]) -> None:
+
+    num_iterations = 1000
+    num_dimensions = 20
+    grid_shape = (50, 50)
+    batch_size = 36
+    sigma_g = 0.5
+    minval = -5.12
+    maxval = 5.12
+    min_bd = -5.12 * 0.5 * num_dimensions
+    max_bd = 5.12 * 0.5 * num_dimensions
+    pool_size = 3
+
+    def sphere_scoring(x: jnp.ndarray) -> jnp.ndarray:
+        return -jnp.sum((x + minval * 0.4) * (x + minval * 0.4), axis=-1)
+
+    fitness_scoring = sphere_scoring
+
+    def clip(x: jnp.ndarray) -> jnp.ndarray:
+        in_bound = (x <= maxval) * (x >= minval)
+        return jnp.where(condition=in_bound, x=x, y=(maxval / x))
+
+    def _behavior_descriptor_1(x: jnp.ndarray) -> jnp.ndarray:
+        return jnp.sum(clip(x[: x.shape[-1] // 2]))
+
+    def _behavior_descriptor_2(x: jnp.ndarray) -> jnp.ndarray:
+        return jnp.sum(clip(x[x.shape[-1] // 2 :]))
+
+    def _behavior_descriptors(x: jnp.ndarray) -> jnp.ndarray:
+        return jnp.array([_behavior_descriptor_1(x), _behavior_descriptor_2(x)])
+
+    def scoring_function(x: jnp.ndarray) -> Tuple[Fitness, Descriptor, Dict]:
+        scores, descriptors = fitness_scoring(x), _behavior_descriptors(x)
+        return scores, descriptors, {}
+
+    def scoring_fn(
+        x: jnp.ndarray, random_key: RNGKey
+    ) -> Tuple[Fitness, Descriptor, ExtraScores, RNGKey]:
+        fitnesses, descriptors, extra_scores = jax.vmap(scoring_function)(x)
+        return fitnesses, descriptors, extra_scores, random_key
```

Comment:
Couldn't we use directly the `rastrigin_proj_scoring_function` to make things shorter, and avoid code replication?

Reply:
I perfectly agree. I thought we could keep it for a further PR that will unify all the notebooks and tests.
See issue #95 

Would you be ok to wait for another PR to do it?

Reply:
Yes I agree :)